optimum/intel/ipex/inference.py

```diff
@@ -19,6 +19,54 @@
 if is_ipex_available():
     import intel_extension_for_pytorch as ipex
 
+def sparse_model_config(model_config):
+    embedding_size = None
+    if hasattr(model_config, 'hidden_size'):
+        embedding_size = model_config.hidden_size
+    elif hasattr(model_config, 'n_embed'):
+        embedding_size = model_config.n_embed
+    elif hasattr(model_config, 'n_embd'):
+        embedding_size = model_config.n_embd
+
+    num_head = None
+    if hasattr(model_config, 'num_attention_heads'):
+        num_head = model_config.num_attention_heads
+    elif hasattr(model_config, 'n_head'):
+        num_head = model_config.n_head
+
+    if embedding_size is None or num_head is None or num_head == 0:
+        raise ValueError("Check the model config")
+
+    num_embedding_size_per_head = int(embedding_size/num_head)
+    num_layer = model_config.n_layer
+
+    return num_layer, num_head, num_embedding_size_per_head
+
+def prepare_jit_inputs(inputs, model_pipeline, dtype):
+
+    input_length = 1 if isinstance(inputs, str) else len(inputs)
+    if model_pipeline.tokenizer.pad_token is None:
+        model_pipeline.tokenizer.pad_token = model_pipeline.tokenizer.eos_token
+    dummy_input = model_pipeline.tokenizer.batch_encode_plus(inputs, return_tensors="pt", padding=True)
+    num_block_layers, num_head, num_embedding_size_per_head = sparse_model_config(model_pipeline.model.config)
+    num_attention_heads = int(num_head*input_length)
+    if model_pipeline.model.config.architectures[0].startswith("Bloom"):
+        past_key_values = tuple(tuple([tuple([torch.zeros(num_attention_heads, 1, num_embedding_size_per_head, 1).to(dtype), \
+                                            torch.zeros(num_attention_heads, 1, num_embedding_size_per_head).to(dtype)])]) * num_block_layers)
+    else:
+        past_key_values = tuple(tuple([tuple([torch.zeros(input_length, num_head, 1, num_embedding_size_per_head).to(dtype), \
+                                            torch.zeros(input_length, num_head, 1, num_embedding_size_per_head).to(dtype)])]) * num_block_layers)
+    dummy_input['attention_mask'] = torch.cat([torch.zeros(dummy_input['attention_mask'].shape[0], 1).to(dummy_input['attention_mask'].dtype), \
+                                                dummy_input['attention_mask']], -1)
+    if model_pipeline.task == 'text-generation':
+        jit_inputs = tuple([dummy_input['input_ids'], past_key_values, dummy_input['attention_mask']])
+    elif model_pipeline.task == 'text2text-generation':
+        # need to do ...
+        jit_inputs = None
+    else:
+        jit_inputs = tuple([dummy_input['input_ids'], dummy_input['attention_mask']])

```

Comment:
Can you replace the single by double quotes everywhere for uniformity ? 
```suggestion
        jit_inputs = tuple([dummy_input["input_ids"], dummy_input["attention_mask"]])

```